processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -328,13 +329,14 @@ public IncrementalIndex(
       capabilities.setType(type);
       columnCapabilities.put(entry.getKey(), capabilities);
     }
-    this.dimValues = new DimensionHolder();
+    this.dimValues = new DimensionHolder(32 + (dimensions.size() + metrics.length) << 2);
```

Comment:
It's my laziness, wanting the array final. I'll check performance difference with proper increasing-array semantic.